controllers/storageclassclaimsecret/storageclassclaimsecret_controller.go

```diff
@@ -0,0 +1,665 @@
+				parts := strings.Split(annotation, "/")
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Namespace: parts[0],
+						Name:      parts[1],
+					},
+				}}
+			}
+			return []reconcile.Request{}
+		})
+	// Compose a predicate that is an OR of the specified predicates
+	//secretsPredicate := util.ComposePredicates(
+	//	predicate.GenerationChangedPredicate{},
+	//	util.MetadataChangedPredicate{},
+	//)
+
+	// As we are not setting the Controller OwnerReference on the ceph
+	// resources we are creating as part of the StorageClassClaim, we need to
+	// set IsController to false to get Reconcile Request of StorageClassClaim
+	// for the owned ceph resources updates.
+	enqueueForNonControllerOwner := &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.StorageClassClaim{}, IsController: false}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}, builder.WithPredicates(claimPredicate)).
+		Watches(&source.Kind{Type: &rookCephv1.CephBlockPool{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephFilesystemSubVolumeGroup{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephClient{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &storagev1.StorageClass{}}, enqueueStorageConsumerRequest).
+		Watches(&source.Kind{Type: &snapapi.VolumeSnapshotClass{}}, enqueueStorageConsumerRequest).
+		Complete(r)
+}
+
+func (r *StorageClassClaimSecretReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim Secret controller in Provider Mode")
+
+	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimInitializing
+
+	gvk, err := apiutil.GVKForObject(&v1alpha1.StorageConsumer{}, r.Client.Scheme())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get gvk for consumer  %w", err)
+	}
+	// reading storageConsumer Name from storageClassClaimSecret ownerReferences
+	ownerRefs := r.storageClassClaimSecret.GetOwnerReferences()
+	for i := range ownerRefs {
+		if ownerRefs[i].Kind == gvk.Kind {
+			r.storageConsumer = &v1alpha1.StorageConsumer{}
+			r.storageConsumer.Name = ownerRefs[i].Name
+			r.storageConsumer.Namespace = r.OperatorNamespace
+			break
+		}
+	}
+	if r.storageConsumer == nil {
+		return reconcile.Result{}, fmt.Errorf("no storage consumer owner ref on the storage class claim")
+	}
+
+	if err := r.get(r.storageConsumer); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if r.storageClassClaim.Spec.Type == "blockpool" {
+		r.cephBlockPool = &rookCephv1.CephBlockPool{}
+		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
+		r.cephBlockPool.Namespace = r.OperatorNamespace
+	} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
+		r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s", r.storageConsumer.Name)
```

Comment:
subvolumegroup name is changed in 4.12 not sure we are handing that in this PR

Reply:
Fixed in rebase